checks.d/mysql.py

```diff
@@ -806,13 +806,18 @@ def _is_innodb_engine_enabled(self, db):
         # Whether InnoDB engine is available or not can be found out either
         # from the output of SHOW ENGINES or from information_schema.ENGINES
         # table. Later is choosen because that involves no string parsing.
-        with closing(db.cursor()) as cursor:
-            cursor.execute(
-                "select engine from information_schema.ENGINES where engine='InnoDB'")
+        try:
+            with closing(db.cursor()) as cursor:
+                cursor.execute(
+                    "select engine from information_schema.ENGINES where engine='InnoDB' and \
+                    support not like 'no' and support not like 'disabled'"
```

Comment:
@olivielpeau I'm not totally sure about the possible "Disabled", "DISABLED", "disabled" combinations, that's what I did it this way....